packages/components/atoms/f-toggle/src/components/Toggle.vue

```diff
@@ -0,0 +1,223 @@
+<template>
+    <label
+        :for="uniqueId"
+        data-test-id="toggle-component"
+        :class="[$style['c-toggle'],
+                 {
+                     [$style['c-toggle--checked']]: isChecked,
+                     [$style['c-toggle--disabled']]: $attrs.disabled,
+                     [$style['c-toggle--modeRTL']]: isModeRightToLeft
+                 }]">
+        <input
+            :id="uniqueId"
+            role="switch"
+            type="checkbox"
+            v-bind="$attrs"
+            :class="$style['c-toggle-input']"
+            :aria-checked="isChecked"
+            :aria-labelledby="ariaLabelledBy"
+            :aria-describedby="ariaDescribedBy"
+            @change="toggleOption">
+
+        <div :class="$style['c-toggle-control']">
+            <check-icon
+                v-if="isChecked"
+                :class="$style['c-toggle-icon']"
+                aria-hidden />
+        </div>
+    </label>
+</template>
+
+<script>
+import { CheckIcon } from '@justeattakeaway/pie-icons-vue';
+import { v4 as uuid } from 'uuid';
+
+export default {
+    name: 'FToggle',
+
+    components: {
+        CheckIcon
+    },
+
+    inheritAttrs: false,
+
+    props: {
+        isModeRightToLeft: {
+            type: Boolean,
+            default: false
+        },
+
+        ariaLabelledBy: {
+            type: String,
+            default: ''
+        },
+
+        ariaDescribedBy: {
+            type: String,
+            default: ''
+        }
+    },
+
+    computed: {
+        uniqueId () {
+            const idArray = ['toggle'];
+
+            if (this.$attrs.name) idArray.push(this.$attrs.name);
+            if (this.$attrs.id) idArray.push(this.$attrs.id);
+
+            idArray.push(this.uniqueIdentifierId);
+            return idArray.join('-');
+        },
+
+        isChecked () {
+            return this.$attrs.checked;
+        }
+    },
+
+    created () {
+        this.uniqueIdentifierId = uuid();
+    },
+
+    methods: {
+        toggleOption (event) {
+            this.$emit('update', event.target.checked);
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+@use '@justeat/fozzie/src/scss/fozzie' as f;
+
+$toggle-colour: f.$color-interactive-form;
+$toggle-colour--checked: f.$color-interactive-brand;
+$toggle-colour--disabled: f.$color-disabled-01;
+
+$toggle-width: 48px;
+$toggle-height: 24px;
+$toggle-control-size: 20px;
+$toggle-padding: 2px;
+$toggle-radius: f.$radius-rounded-e;
+$toggle-icon-size: 14px;
+
+$toggle-translation: translateX(calc($toggle-width - $toggle-control-size - 2 * $toggle-padding));
+
+@mixin toggle-transition ($property) {
+    transition-property: $property;
+    transition-duration: 0.15s;
+    transition-timing-function: cubic-bezier(0.4, 0, 0.9, 1);
+    transition-delay: 0s;
+}
+
+.c-toggle {
+    display: flex;
+
+    @include toggle-transition(background-color);
```

Comment:
I'd wrap all animation in a media query to check the user doesn't have reduced motion enabled. This way people do not like animations or are caused issues by them can opt out.

A basic example can be seen [here](https://web.dev/prefers-reduced-motion/#working-with-the-media-query-from-css-and-javascript-contexts) :)